### PR TITLE
[RFR] Move tile to opposite (follows #3)

### PR DESCRIPTION
**Pull request by HALLERPierre**
[trello](https://trello.com/c/NDQPAq4r/2-as-fred-i-want-to-move-a-cube-05)

# TODO 

- [x] Move row
- [x] Move col
- [x] Move tile
- [x] Move to opposite
- [x] Ask for user input
- [x] Add tests

# Result
Peek 2019-05-28 11-53

### CODE REVIEW

**Review by zyhou — CHANGES_REQUESTED** [review]
Need rebase.

On some function, you mutate the param of the function. Do immutable function.

File: src/game.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from moves import is_movable_tile, get_opposite_tile, move_tile
+from board import create_board, print_board
+import os
+
+clear = lambda : os.system('clear')
+
+def get_instructions(board, player_value=0):
```
Comment (Luwangel) [inline src/game.py]: This method's name is not so clear. Could you find a better name?

File: src/game.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+def play():
+    clear()
+    player_team = 1
+    board = create_board()
+    while True:
+        instructions = get_instructions(board, player_team)
+        print_board(board, instructions)
+        instruction = int(input('Which tile do you want to move ?'))
+        # check_input()
```
Comment (Luwangel) [inline src/game.py]: Comment spotted

**Review by zyhou — APPROVED** [review]

File: src/board.py [hunk 3]
```diff
@@ -1,18 +1,58 @@
...
-            symbol = get_symbol(tile)
-            print('[' + symbol + ']', end='')
-        print('')
+def print_board(board, instructions={}):
+    for x in range(len(board)):
+        for y in range(len(board[x])):
+            symbol = get_symbol(board[x][y])
+            instruction = instructions[(x, y)] if (x, y) in instructions else None
+            tileText = f'\t[{symbol}]' if instruction is None else f'\t[{symbol}]({instruction})'
+            print(f'{tileText}', end='\t')
```
Comment (Kmaschta) [inline src/board.py:22]: No need of literal string if you just send a variable.
Just write `print(tileText, end='\t')`

File: src/moves.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+from constants import N_ROWS, N_COLS
+
+
+def move_row(board, row, y_start, y_end, value):
+    board_copy = deepcopy(board)
+
+    step = -1 if y_end > y_start else 1
+    index_start = y_start - 1 if y_end > y_start else y_start + 1
+    for y in range(y_end, index_start, step):
+        prevVal = board_copy[row][y]
```
Comment (Kmaschta) [inline src/moves.py]: Camel case instead of sneaky_case

File: src/moves.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    Returns:
+        The game board
+    """
+    board_copy = deepcopy(board)
+
+    (x, y) = pos_start
+    (x_end, y_end) = pos_end
+
+    if not is_movable_tile(x, y):
+        raise Exception("Can't move a tile that is in the center of the board")
```
Comment (Kmaschta) [inline src/moves.py:55]: It's better to have custom exception classes, in order to catch them later if needed, and not catch every other exceptions

File: src/moves.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        GameException: Can't move a tile that is in the corner (for the moment)
+
+    Returns:
+        tupplet -- x & y coords of the opposing tile
+    """
+
+    if not is_movable_tile(x, y):
+        raise Exception("Can't move a tile that is in the center of the board")
+
+    if x == y or x == 0 and y == 4 or x == 4 and y == 0:
```
Comment (Kmaschta) [inline src/moves.py]: Why 4 here? Does this number depends of the constant?

Comment (HALLERPierre) [reply src/moves.py]: Yes, I should add the constants here.

File: src/moves.py (hunk 5 above)
Comment (HALLERPierre) [reply src/moves.py:55]: Yes, I'll do it.

**Pull request merged**